### ISSUE TITLE
Add the ability to filter events for versions 'less than' a specific version in the inbound filter

### ISSUE THREAD

**Issue by PhilHannent** [getsentry/sentry#43941]
### Problem Statement

A custom filter for a given text string is a risky addition to add as you are blocking the ability to detect regressions. It would be better to filter out older versions of the software from bug reports. The existing filter only support glob matching, however this would then mean that with each release of the software we need to append the older version to that list. The list would then become very long and need to be maintained.

https://docs.sentry.io/product/data-management-settings/filtering/

### Solution Brainstorm

My initial thought would be that rather than glob, there needs to be a filter where we can ignore versions less than a certain release. For example: < 7.7.8

It might also be nice to have an API endpoint which we can put into our CI/CD system which would update the field of the current release version.

The added complexity is that release channels would have a different filter. A beta channel could be on v7.8.0 whereas a production would still be on 7.7.10. Developers would want to see both.

Perhaps a simpler solution would be a switch which when on only allowed the most recent version of that channels bugs.

**Comment by scttcper** [comment]
@PhilHannent we've heard that people want more options when filtering by releases. Eg "last 2 releases". Which part of the product specifically are you referring to? Alerts? Issue search?
Thanks!

**Comment by PhilHannent** [comment]
This issue came up due to the addition of attachments in our Qt/C++ integration. Previously, we had an in house system and we had a constant for a version we didn't care about (so we could see the previous couple of versions). 

However, we are looking to be more forceful with getting users to upgrade to the latest version, so anything less than what is in the production release is not worth spending time on. I appreciate we might miss problems, but statistically, if the problem hasn't been fixed it will turn up in the latest eventually.

A big factor for us turning on alerts is the signal to noise ratio, we really want to keep the noise down, so that developers are notified of something truly new and especially to jump on problems in a new release.

**Comment by meotimdihia** [comment]
Yes, it is weird, Sentry should have this feature. Events that do not belong to the latest version should be filtered
The browsers always cache the old version, old events are still sent for days.

**Comment by leo-liu1** [comment]
Would love to see this feature added. We have errors resurface in Sentry from really old versions and continue to fire alerts on our channel. We don't find these errors to be really useful since these would usually be patched in future versions.

Currently we do glob pattern matching to filter out these old versions, but it would be nice to have something more expressive. Alternatively, something that allows us to only surface errors from X releases ago.

**Comment by alfredocambera-changeorg** [comment]
I'd like to see this issue fixed and I'm getting issues created for releases that we no longer support
